File: src/main/java/net/dumbdogdiner/dogcore/commands/snoop/InvSeeCommand.java
package net.dumbdogdiner.dogcore.commands.snoop;

import dev.jorel.commandapi.annotations.Command;
import dev.jorel.commandapi.annotations.Default;
import dev.jorel.commandapi.annotations.Permission;
import dev.jorel.commandapi.annotations.arguments.APlayerArgument;
import net.dumbdogdiner.dogcore.Permissions;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

@Command("invsee")
@Permission(Permissions.SNOOP)
public final class InvSeeCommand {
    private InvSeeCommand() { }

    @Default
    public static void invSee(final @NotNull Player sender, final @NotNull @APlayerArgument Player player) {
        sender.openInventory(player.getInventory());
    }
}

File: src/main/kotlin/net/dumbdogdiner/dogcore/commands/InvSeeCommand.kt
package net.dumbdogdiner.dogcore.commands

import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerArgument
import dev.jorel.commandapi.kotlindsl.playerExecutor
import net.dumbdogdiner.dogcore.Permissions
import org.bukkit.entity.Player

fun invSeeCommand() = commandAPICommand("invsee") {
    playerArgument("player")
    withPermission(Permissions.SNOOP)

    playerExecutor { sender, args ->
        // TODO we could look into offline player support
        // TODO does not show armor slots/offhand
        val player = args["player"] as Player
        sender.openInventory(player.inventory)
    }
}
